bluepyopt/ephys/create_acc.py

```diff
@@ -0,0 +1,451 @@
+            for param in mech_params:
+                assert param.name in arb_mech['globals'] or \
+                       param.name in arb_mech['ranges']
+            mech_params_dict = dict(mech_params)
+            arb_mech_name = mech_name + '/' + ','.join(
+                [p + '=' + mech_params_dict[p] for p in arb_mech['globals']])
+            arb_mech_params = [mech_param for mech_param in mech_params
+                               if mech_param.name not in arb_mech['globals']]
+            return (arb_mech_name, arb_mech_params)
+
+
+def _arb_nmodl_global_translate_local(params):
+    ret = []
+    for loc, mechs in params:
+        ret.append((loc, [_arb_nmodl_global_translate(*mech)
+                          for mech in mechs]))
+    return ret
+
+
+def _read_templates(template_dir, template_filename):
+    """Expand Jinja2 template filepath with glob and
+     return dict of target filename -> parsed template"""
+    if template_dir is None:
+        template_dir = os.path.abspath(
+            os.path.join(
+                os.path.dirname(__file__),
+                'templates'))
+
+    template_paths = glob(os.path.join(template_dir,
+                                       template_filename))
+
+    templates = dict()
+    for template_path in template_paths:
+        with open(template_path) as template_file:
+            template = template_file.read()
+            name = os.path.basename(template_path)
+            if name.endswith('.jinja2'):
+                name = name[:-7]
+            if name.endswith('_template'):
+                name = name[:-9]
+            if '_' in name:
+                name = '.'.join(name.rsplit('_', 1))
+            templates[name] = jinja2.Template(template)
+    return templates
+
+
+def create_acc(mechs,
+               parameters,
+               morphology=None,
+               ignored_globals=(),
+               replace_axon=None,
+               template_name='CCell',
+               template_filename='acc/*_template.jinja2',
+               disable_banner=None,
+               template_dir=None,
+               custom_jinja_params=None):
+    '''return a dict with strings containing the rendered JSON/ACC templates
+
+    Args:
+        mechs (): All the mechs for the decor template
+        parameters (): All the parameters in the decor/label-dict template
+        morphology (str): Name of morphology
+        ignored_globals (iterable str): Skipped NrnGlobalParameter in decor
+        replace_axon (str): String replacement for the 'replace_axon' command.
+        Only False is supported at the moment.
+        template_filename (str): file path of the cell.json , decor.acc and
+        label_dict.acc jinja2 templates (with wildcards expanded by glob)
+        template_dir (str): dir name of the jinja2 templates
+        custom_jinja_params (dict): dict of additional jinja2 params in case
+        of a custom template
+    '''
+
+    if morphology[-4:] not in ['.swc', '.asc']:
```

Comment:
is morphology guaranteed to be lower case?

Reply:
No, but this should be solved in de52dea.